### PR TITLE
ddtrace/tracer: have tracer.Start return an error

### PR DESCRIPTION
**Pull request by JianyiGao**
### What does this PR do?
Added err type in tacer.Start() return value
Proposal: https://github.com/DataDog/dd-trace-go/issues/540
Initial PR: https://github.com/DataDog/dd-trace-go/pull/2231 Moving to this PR as we want to point against v2-dev

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -82,11 +82,10 @@ require (
 	github.com/vektah/gqlparser/v2 v2.5.8
 	github.com/zenazn/goji v1.0.1
 	go.mongodb.org/mongo-driver v1.12.1
-	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.44.0
-	go.opentelemetry.io/otel v1.18.0
-	go.opentelemetry.io/otel/trace v1.18.0
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.42.0
+	go.opentelemetry.io/otel v1.16.0
+	go.opentelemetry.io/otel/trace v1.16.0
```
Comment (darccio) [inline go.mod:87]: @JianyiGao Why are these `go.mod` updates here?

**Review by darccio — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (JianyiGao) [reply go.mod:87]: These changes come from commit [114dba2](https://github.com/DataDog/dd-trace-go/pull/2314/commits/114dba2dbf5801288de0cc7c1ac61c9bff83710f) merged v2-main

Comment (dianashevchenko) [reply go.mod]: @darccio  We checked and removed those, so no updates now in go.mod

**Comment by darccio** [comment]
Looks good. Merged!

**Pull request merged**